scripts/generate_sudo_list.py

```diff
@@ -0,0 +1,64 @@
+"""Helper script to generate sudo list doc."""
+import filecmp
+import sys
+import tempfile
+from pathlib import Path
+
+import yaml
+
+
+def collect_sudo_commands():
+    """Collect all ansible cmds that require sudo."""
+    network_runner_path = Path("quipucords/scanner/network/runner")
+    list_cmds = []
+    for playbook in network_runner_path.glob("roles/*/tasks/main.yml"):
+        for task in yaml.safe_load(playbook.open()):
+            if "user_has_sudo" in task.get("when", ""):
+                list_cmds.append(task.get("raw", ""))
+    return list_cmds
+
+
+def generate_sudo_cmds_document():
+    """Generate a document listing commands that require sudo."""
+    list_cmds = collect_sudo_commands()
+    document = ""
+    document += "\n".join(cmd for cmd in list_cmds if cmd.strip())
+    return document
+
+
+def create_document_file(file_path):
+    """Create a txt file containing a list of commands that require sudo."""
+    document = generate_sudo_cmds_document()
+    file = Path(file_path)
+    file.write_text(document)
+
+
+def compare_sudo_cmds_lists(file_path):
+    """Compare the most updated sudo list with an existing one."""
+    generated_doc = generate_sudo_cmds_document()
+
+    with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+        temp_file.write(generated_doc.encode())
+        try:
+            if filecmp.cmp(temp_file.name, file_path):
+                sys.stdout.write("The sudo command list document is up to date.")
+            else:
+                sys.stderr.write(
+                    "Error: The sudo command list document has been modified.\n"
+                )
+        except FileNotFoundError:
+            sys.stderr.write(f"Error: {file_path} has not been found.\n")
+        finally:
+            Path(temp_file.name).unlink()
+
+
+functions = {
+    "docs": create_document_file,
+    "compare": compare_sudo_cmds_lists,
+}
+
+if __name__ == "__main__":
+    selected_function = functions[sys.argv[1]]
+    function_arguments = sys.argv[2:]
+
+    sys.exit(selected_function(*function_arguments))
```

Comment:
[optional] I think we should establish a pattern for "quipucords scripts" - with this being the first one.

Requirements for quipucords scripts:
- should be [django management commands](https://docs.djangoproject.com/en/4.2/howto/custom-management-commands/)
- they should be covered in tests as the rest of the application
- exceptions can be made for some specific purpose commands, like celery entrypoint or a custom script that should run independently from quipucords dependencies.

Thoughts @abellotti & @infinitewarp ?

Reply:
I think for this purpose, a Django management command would be overkill. Not doing DB, etc.  This is more of a static analysis/linting/man page thing, so a Makefile entry point sounds about right.

Reply:
I agree with @abellotti. As discussed earlier today at standup, a Django management command might be overkill for this script since it's not using our models or other internal quipucords logic. I would not strongly recommend it in this case.

Mirek also pointed out that making this a Django management command would require the user to have the full quipucords source and config locally installed in order to use this script. Because we are committing the output to the repo and docs anyway, the user should not need to run this script directly, but we should not impose that kind of impediment unless actually necessary.